backend/hyper/hyper.go

```diff
@@ -41,6 +41,17 @@ func NewStorage(db *sqlitex.Pool, log *zap.Logger) *Storage {
 	}
 }
 
+// Exec allows to execute raw SQLite write operations.
+func (bs *Storage) Exec(ctx context.Context, fn func(conn *sqlite.Conn) error) (err error) {
```

Comment:
Please don't do this. The whole point of this abstraction avoid leaking raw SQLite connections for mutations, that's why there was only Query method on storage.

Reply:
Yes, but the query is read only so I cannot insert records. I then created a function inside hyper.go to insert the record in the trusted table. But then I realized that maybe that was not a task of the hyper layer, that should be abstracted in some way. So I ended up opening the door to inject malicious queries :)

Reply:
Changed